solver/solver_test.go

```diff
@@ -0,0 +1,23 @@
+package solver
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStats_CreateOps(t *testing.T) {
+	var a AtomicInt32Counter
+	var wg sync.WaitGroup
+
+	wg.Add(10)
+	for i := 0; i < 10; i++ {
+		go func() {
+			defer wg.Done()
+			a.Increment(int32(1))
+		}()
+	}
```

Comment:
optional: too see any meaningful risk of race condition occuring, you probably need each of the goroutines to increment the counter 1000 times in a loop.

Reply:
This is really to trigger a data race bug with `go test -race`.
If we disable locking, this reliably produces the race:
```
go test -race
==================
WARNING: DATA RACE
Read at 0x00c000156240 by goroutine 9:
  github.com/kong/deck/solver.(*AtomicInt32Counter).Increment()
      /home/hbagdi/deck/solver/solver.go:31 +0x6f
  github.com/kong/deck/solver.TestStats_CreateOps.func1()
      /home/hbagdi/deck/solver/solver_test.go:18 +0x66

Previous write at 0x00c000156240 by goroutine 8:
  github.com/kong/deck/solver.(*AtomicInt32Counter).Increment()
      /home/hbagdi/deck/solver/solver.go:31 +0x84
  github.com/kong/deck/solver.TestStats_CreateOps.func1()
      /home/hbagdi/deck/solver/solver_test.go:18 +0x66

Goroutine 9 (running) created at:
  github.com/kong/deck/solver.TestStats_CreateOps()
      /home/hbagdi/deck/solver/solver_test.go:16 +0xf2
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1194 +0x202

Goroutine 8 (finished) created at:
  github.com/kong/deck/solver.TestStats_CreateOps()
      /home/hbagdi/deck/solver/solver_test.go:16 +0xf2
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1194 +0x202
==================
--- FAIL: TestStats_CreateOps (0.00s)
    testing.go:1093: race detected during execution of test
FAIL
exit status 1
FAIL    github.com/kong/deck/solver     0.008s

```